test/run_tests.py

```diff
@@ -96,68 +112,110 @@ def http_server():
                for opt in ROBOT_OPTIONS]
-    if env.TRAVIS:
-        options.extend(get_travis_conf(browser))
-    command = runner + options + cli_options + [env.ACCEPTANCE_TEST_DIR]
-    log_start(command)
+    if sauceusername and saucekey:
+        options.extend(get_sauce_conf(browser, sauceusername, saucekey))
+    command = runner
+    if suite:
+        command += ['-s', suite]
+    command += options + [env.ACCEPTANCE_TEST_DIR]
+    log_start(command, sauceusername, saucekey)
     syslog = os.path.join(env.RESULTS_DIR, 'syslog.txt')
-    # Running tests as an external process, not using `robot_cli`, to allow
-    # using different interpreter that is used for running this script.
-    subprocess.call(command, env=dict(os.environ, ROBOT_SYSLOG_FILE=syslog))
-
-
-def get_travis_conf(browser):
-    conf = [
-        '--variable', 'SAUCE_USERNAME:{}'.format(env.SAUCE_USERNAME),
-        '--variable', 'SAUCE_ACCESS_KEY:{}'.format(env.SAUCE_ACCESS_KEY)
-    ]
-    if browser == 'firefox':
-        conf.extend([
-            '--variable',
-            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
-                env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY
-            )
-        ])
-    else:
-        conf.extend([
-            '--variable',
-            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
-                env.TRAVIS_JOB_NUMBER, browser
-            )
-        ])
-    return conf
+    subprocess.call(
+        command, env=dict(os.environ, ROBOT_SYSLOG_FILE=syslog)
+    )
 
 
-def log_start(command_list):
+def log_start(command_list, sauceusername, saucekey):
     command = subprocess.list2cmdline(command_list)
-    for hidden in [env.SAUCE_USERNAME, env.SAUCE_ACCESS_KEY]:
+    for hidden in [sauceusername, saucekey]:
         if hidden:
             command = command.replace(hidden, '*' * len(hidden))
     print()
     print('Starting test execution with command:')
     print(command)
 
 
-def process_output(browser, cli_options):
+def get_sauce_conf(browser, sauceusername, saucekey):
+    if browser == 'chrome' and env.TRAVIS:
+        conf = []
+    else:
+        conf = [
+            '--variable', 'SAUCE_USERNAME:{}'.format(sauceusername),
+            '--variable', 'SAUCE_ACCESS_KEY:{}'.format(saucekey),
+            '--variable',
+            'REMOTE_URL:http://{}:{}@ondemand.saucelabs.com:80/wd/hub'.format(
+                sauceusername, saucekey
+            ),
+            '--variable',
+            'DESIRED_CAPABILITIES:build:{0}-{1},tunnel-identifier:{0}'.format(
+                env.TRAVIS_JOB_NUMBER, browser
+            )
+        ]
+    return conf
+
+
+def process_output(browser, suite):
     print('Verifying results...')
     output = os.path.join(env.RESULTS_DIR, 'output.xml')
     robotstatuschecker.process_output(output, verbose=False)
     options = [opt.format(browser=browser) for opt in REBOT_OPTIONS]
+    if suite:
+        options += ['-s', suite]
```

Comment:
This might still valid, but the code has changed quite a look. Please take a new look and send comment if something is needed.